config/routes.rb

```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  # --- GET ---
+  # users
+  get "users", to: "users#index"
+  get "users/:id", to: "users#show"
+  # users posts
+  get "users/:user_id/posts", to: "posts#index"
+  get "users/:user_id/posts/:id", to: "posts#show"
```

Comment:
- [ ] [OPTIONAL] It will be better to use nested routes to make your code clean and to not write more code in routes for every action like the follwing 
```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
end
```

Reply:
Thank you for the advice, I will update it